### ISSUE TITLE
PyTorch model: Error on save

### ISSUE THREAD

**Issue by ankxyz** [iterative/mlem#288]
**Code**:

```python
import torch
import math
from mlem.api import save


x = torch.linspace(-math.pi, math.pi, 2000)
y = torch.sin(x)

p = torch.tensor([1, 2, 3])
xx = x.unsqueeze(-1).pow(p)

model = torch.nn.Sequential(
    torch.nn.Linear(3, 1),
    torch.nn.Flatten(0, 1)
)
loss_fn = torch.nn.MSELoss(reduction='sum')
learning_rate = 1e-6

for t in range(2000):

    y_pred = model(xx)
    loss = loss_fn(y_pred, y)

    if t % 100 == 99:
        print(t, loss.item())

    model.zero_grad()
    loss.backward()

    with torch.no_grad():
        for param in model.parameters():
            param -= learning_rate * param.grad

linear_layer = model[0]

print(f'Result: y = {linear_layer.bias.item()} + {linear_layer.weight[:, 0].item()} x + {linear_layer.weight[:, 1].item()} x^2 + {linear_layer.weight[:, 2].item()} x^3')


save(
    model,
    'torch_model',
    sample_data=xx,
    description='Example of pytorch model'
)
```

---

**Environment**:
- *OS*: Ubuntu 20.04
- *Python*: 3.8.10
- *Python package manager*: pip 22.1.2
- *mlem*: 0.2.2
- *pytorch*: 1.11.0

**Error**:

```
Traceback (most recent call last):
  File "train.py", line 39, in <module>
    save(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 99, in save
    meta.dump(path, fs=fs, project=project, index=index, external=external)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 508, in dump
    self._write_meta(location, index)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 243, in _write_meta
    safe_dump(self.dict(), f)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/__init__.py", line 269, in safe_dump
    return dump_all([data], stream, Dumper=SafeDumper, **kwds)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/__init__.py", line 241, in dump_all
    dumper.represent(data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 27, in represent
    node = self.represent_data(data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 48, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 207, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 118, in represent_mapping
    node_value = self.represent_data(item_value)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 48, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 199, in represent_list
    return self.represent_sequence('tag:yaml.org,2002:seq', data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 92, in represent_sequence
    node_item = self.represent_data(item)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 48, in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 207, in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 118, in represent_mapping
    node_value = self.represent_data(item_value)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 58, in represent_data
    node = self.yaml_representers[None](self, data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/yaml/representer.py", line 231, in represent_undefined
    raise RepresenterError("cannot represent an object", data)
yaml.representer.RepresenterError: ('cannot represent an object', '1.11.0+cu102')
```

---

*UPD*: Another `PyTorch` version tried:
*PyTorch*: 1.10.0

**Error**:
```
Traceback (most recent call last):
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 34, in get_object_metadata
    return MlemData.from_data(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 675, in from_data
    data_type = DataType.create(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/data_type.py", line 74, in create
    return DataAnalyzer.analyze(obj, **kwargs).bind(obj)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 135, in _find_hook
    raise HookNotFound(
mlem.core.errors.HookNotFound: No suitable DataHook for object of type "Sequential". Registered hooks: [<class 'mlem.core.data_type.PrimitiveType'>, <class 'mlem.core.data_type.OrderedCollectionHook'>, <class 'mlem.core.data_type.DictType'>, <class 'mlem.contrib.numpy.NumpyNumberType'>, <class 'mlem.contrib.numpy.NumpyNdarrayType'>, <class 'mlem.contrib.torch.TorchTensorDataType'>]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 39, in <module>
    save(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 96, in save
    meta = get_object_metadata(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 38, in get_object_metadata
    return MlemModel.from_obj(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 611, in from_obj
    mt = ModelAnalyzer.analyze(model, sample_data=sample_data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/model.py", line 298, in analyze
    return super().analyze(obj, sample_data=sample_data, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/contrib/torch.py", line 150, in process
    PREDICT_METHOD_NAME: Signature.from_method(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/model.py", line 164, in from_method
    returns = DataAnalyzer.analyze(result)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/contrib/torch.py", line 78, in process
    dtype=str(obj.dtype)[len(obj.dtype.__module__) + 1 :],
AttributeError: 'torch.dtype' object has no attribute '__module__'
```

**Comment by mike0sv** [comment]
This should be already fixed in #281, can you try installing from `main`? Or wait for release, we are planning to do it soon

**Comment by ankxyz** [comment]
> This should be already fixed in #281, can you try installing from `main`? Or wait for release, we are planning to do it soon

Thanks! I tried it (installed from *main*) and it works. But it works just with `torch==1.11.0` (latest). With `torch==1.10.0` the same error:

```
Traceback (most recent call last):
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 34, in get_object_metadata
    return MlemData.from_data(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 675, in from_data
    data_type = DataType.create(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/data_type.py", line 74, in create
    return DataAnalyzer.analyze(obj, **kwargs).bind(obj)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 135, in _find_hook
    raise HookNotFound(
mlem.core.errors.HookNotFound: No suitable DataHook for object of type "Sequential". Registered hooks: [<class 'mlem.core.data_type.PrimitiveType'>, <class 'mlem.core.data_type.OrderedCollectionHook'>, <class 'mlem.core.data_type.DictType'>, <class 'mlem.contrib.numpy.NumpyNumberType'>, <class 'mlem.contrib.numpy.NumpyNdarrayType'>, <class 'mlem.contrib.torch.TorchTensorDataType'>]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 39, in <module>
    save(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 96, in save
    meta = get_object_metadata(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/metadata.py", line 38, in get_object_metadata
    return MlemModel.from_obj(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/objects.py", line 611, in from_obj
    mt = ModelAnalyzer.analyze(model, sample_data=sample_data)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/model.py", line 298, in analyze
    return super().analyze(obj, sample_data=sample_data, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/contrib/torch.py", line 150, in process
    PREDICT_METHOD_NAME: Signature.from_method(
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/model.py", line 164, in from_method
    returns = DataAnalyzer.analyze(result)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/core/hooks.py", line 107, in analyze
    return cls._find_hook(obj).process(obj, **kwargs)
  File "/media/alex/hdd/Dev/progexp/mlem/pytorch-example/.venv/lib/python3.8/site-packages/mlem/contrib/torch.py", line 78, in process
    dtype=str(obj.dtype)[len(obj.dtype.__module__) + 1 :],
AttributeError: 'torch.dtype' object has no attribute '__module__'
```

Is it expected behavior that `mlem` supports just the latest version of `PyTorch`?

**Comment by mike0sv** [comment]
It's better to have latest versions, yes, since we cant test against every version of every framework. But this one should now work with previous versions (I only checked 1.10 though)
